03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,25 +1,51 @@
 const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/async')  //a middleware to not repeat try catch block in every fun
+const { createCustomError } = require('../errors/custom-error')
 
-const getTasks = (req,res)=>{
-    res.send('All tasks')
-}
+//get all tasks
+const getTasks = asyncWrapper (async (req,res)=>{
+    const tasks = await Task.find({})
+    res.status(201).json({tasks})
+})
 
-const getSingleTask = (req,res)=>{
-    res.send('Get single task')
-}
+//get a task
+const getSingleTask = asyncWrapper(async(req,res)=>{
+   
+    const {id: taskID} = req.params;
+    const task = await Task.findOne({_id : taskID})
+    if(!task){
+        return next(createCustomError(`No task with id : ${taskID}`, 404))
+    }
+    res.status(200).json({task})
+   
+})
 
-const createTask= async (req,res)=>{
+//add a new task
+const createTask= asyncWrapper(async (req,res)=>{
     const task= await Task.create(req.body)
-    res.status(201).json({task})
-}
+    res.status(201).json({task})     
+})
 
-const updateTask = (req,res)=>{
-    res.send('update task')
-}
+//update a task
+const updateTask = asyncWrapper(async(req,res)=>{
+    const {id:taskID} = req.params;
+    const task = await Task.findOneAndUpdate({_id : taskID},req.body)
+    if(!task){
+        return next(createCustomError(`No task with id : ${taskID}`, 404))
+    }
+    res.status(200).json({task})
+})
 
-const deleteTask= (req,res)=>{
-    res.send('delete task')
-}
+//remove a task
+const deleteTask= asyncWrapper(async(req,res)=>{
```

Comment:
Missing next as parameter
```suggestion
const getSingleTask = asyncWrapper(async(req,res,next)=>{
```